packages/azure-http-specs/specs/azure/resource-manager/common-properties/mockapi.ts

```diff
@@ -126,3 +126,60 @@ Scenarios.Azure_ResourceManager_CommonProperties_ManagedIdentity_updateWithUserA
     },
     kind: "MockApiDefinition",
   });
+
+Scenarios.Azure_ResourceManager_CommonProperties_Error_get = passOnCode(404, {
+  uri: "/subscriptions/:subscriptionId/resourceGroups/:resourceGroup/providers/Azure.ResourceManager.CommonProperties/confidentialResources/:resourceName",
+  method: "get",
+  request: {
+    params: {
+      subscriptionId: SUBSCRIPTION_ID_EXPECTED,
+      resourceGroup: RESOURCE_GROUP_EXPECTED,
+      resourceName: "confidential",
+      "api-version": "2023-12-01-preview",
+    },
+    status: 404,
```

Comment:
Is it intentional that we define a field `request.status` with an expected response status code?

Reply:
Yes. It's for matching the 404 `response.status`, to pass Spector e2e test.
Spector code: https://github.com/microsoft/typespec/blob/3f7204c0a4e05aa04126ac736f640b2c0047b3b4/packages/spector/src/actions/server-test.ts#L24-L29

Existing usage: https://github.com/microsoft/typespec/blob/657d4a0f1c86acb2a0d31b51aa1d128d2a4d6571/packages/http-specs/specs/authentication/http/custom/mockapi.ts#L26

Reply:
@sarangan12 May need your opinion here. I thought `response.status` should be enough, to match `passOnCode(status)`. Why is `request.status` still in need?

Reply:
Thanks for clarifying